src/state_machine/coordinator/fire.rs

```diff
@@ -249,7 +249,9 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
                     return Ok((None, None));
                 }
                 State::DkgPublicDistribute => {
-                    let packet = self.start_public_shares()?;
+                    // XXX does this ever happen? start_dkg_round should call start_public_shares
+                    //panic!("Got message in DkgPublicDistribute");
+                    let packet = self.start_public_shares(false)?;
```

Comment:
I suppose these comments should be removed, but I'm curious about the back story to the comment here.

Reply:
I noticed that here we don't have a straight call stack from `start_dkg_round` to `start_public_shares`.  I then realized that `DkgPublicDistribute` is an ephemeral state, so we should never be receiving message while in it.  So I added the `panic!` to check, and we aren't.

Reply:
The other `*Distribute` states are also ephemeral, but we do loop back to the state handler before doing anything about them.  `DkgPublicDistribute` does not.

Reply:
I'm tempted to actually uncomment the `panic!` here, since we should never actually get to this branch of the match.

Reply:
Ahh I see. Well, I don't think libraries should ever `panic!` unless the caller is well aware of the possibility of that fact. I suppose I would officially be on notice if this panicked but still.

Reply:
Agreed. That’s why I’m tempted but didn’t actually do it. 

We should just return an error here.

Reply:
Ok now we're returning an error.